msgs-spray-json/src/main/scala/com/thenewmotion/ocpi/msgs/v2_1/OcpiJsonProtocol.scala

```diff
@@ -87,24 +87,37 @@ trait OcpiJsonProtocol extends DefaultJsonProtocol {
 
   implicit val energySourceFormat = jsonFormat2(EnergySource)
   implicit val environmentalImpactFormat = jsonFormat2(EnvironmentalImpact)
-  implicit val energyMixFormat = jsonFormat5(EnergyMix)
+
+  implicit val energyMixFormat = new JsonFormat[EnergyMix] {
+    override def read(json: JsValue) = jsonFormat5(EnergyMix.deserialize).read(json)
+    override def write(obj: EnergyMix): JsValue = jsonFormat5(EnergyMix.apply).write(obj)
+  }
```

Comment:
Probably more efficient to create the jsonFormats once and reuse them rather than creating them every time the read or write methods are called